kart/sqlalchemy/__init__.py

```diff
@@ -153,6 +153,26 @@ def separate_last_path_part(uri):
     return (modified_url, last_part)
 
 
+def truncate_to_path_length(uri, path_length):
+    """
+    Removes any number of path-parts from a URI to truncate it to the required length.
+    Eg, with path_length of 1, it would truncate SCHEME://HOST/DBNAME/DBSCHEMA/TABLE to SCHEME://HOST/DBNAME
+    Or with path_lengh of 2, it would truncate SCHEME://HOST/DBNAME/DBSCHEMA/TABLE to SCHEME://HOST/DBNAME/DBSCHEMA
+    If the uri's path length is already as short or shorter as the given path length, then it has no effect.
+    """
+    url = urlsplit(uri)
+    url_path = PurePosixPath(url.path)
+    current_path_length = len(url_path.parents)
+    if current_path_length <= path_length:
+        return uri
+
+    modified_url_path = str(url_path.parents[-1 - path_length])
```

Comment:
_feels_ like this should just be `-path_length` - the `-1` seems wrong because this is a list of parents.
ie if path_length=1 this should return the last parent ie index -1

Reply:
Dropped this function - realised "startswith" works pretty well here